inveniordm_py/records/resources.py

```diff
@@ -297,3 +304,47 @@ def delete(self):
     def get(self):
         """Get file information."""
         return self._get(FileMetadata)
+
+
+class RecordCommunitiesList(Resource):
+    """Implements a RecordCommunitiesList as a Resource.
+
+    This is the resource that is used to interact with the /api/records/{id_}/communities endpoint.s
+    """
+
+    endpoint = "/records/{id_}/communities"
+
+    def _normalize_data(self, data):
+        """Normalize data to a RecordCommunityMetadata object."""
+        if isinstance(data, RecordCommunityMetadata):
+            _data = data
+        elif isinstance(data, list):
+            _data = RecordCommunityMetadata(communities=data)
+        elif isinstance(data, str):
+            _data = RecordCommunityMetadata(communities=[data])
+        else:
+            raise ValueError(
+                "Data must be a list of communities, a single community id, or a RecordCommunityMetadata object."
+            )
+        return _data
+
+    def add(self, data):
+        """Add communities to a record.
+
+        The data is first normalized, allowing to pass a list of communities, a single community id, or a RecordCommunityMetadata object.
+
+        Usage:
+
+        .. code-block:: python
+
+            communities.add(["com1", "com2"])
+            communities.add("com1")
+            communities.add(RecordCommunityMetadata(communities=["com1", "com2"]))
+        """
+        _data = self._normalize_data(data)
```

Comment:
This behavior is a bit different to other implementations.

E.g. for drafts the usage is `draft.update(data=DraftMetadata({...}))` (note that `data` is a `DraftMetadata` instance)

For communities, it felt better to allow `record.communities.add('blr')` (note that `blr` is just the slug of the community, however an instance of `RecordCommunityMetadata` can also be used)